### DISCUSSION TITLE
Feedback on foundry scripts

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [Cyfrin/foundry-full-course-cu#244] [General]
Hi all!

So many of you who are taking this course are becoming familiar with the scripts feature of foundry.

The foundry team is intent on improving the scripting experience.

What do you think can be done to improve scripting?

**Comment by 0xCalibur** [comment]
definitely having something similar to https://github.com/wighawag/forge-deploy supported natively

**Reply by kartojal** [reply]
As of today, I am really happy with Foundry scripting and using Solidity as a script language, after years fighting with Typescript and Hardhat.

1. Nothing can beat predictability of having all the deployment orchestration in a typed language which is the same language of your smart contracts.
2. At compilation time you can detect errors at constructor parameters if any contract constructor changed.
3. No need to decode and encode Solidity types
4. Perform simulations or do tests against your deployment script to ensure reliability.
5. Integrators can easily import or extend your Solidity deployment script to build on top of your solution. 

What I would improve is the current Etherscan verification algorithm which sometimes fails for some contracts, and allow the developer to modify or change the library deployments step.

**Comment by magnetto90** [comment]
I would like to have an option to use the last deployed contract address in different scripts. I mean, I deploy something and then I can run other scripts with out changing the code but using that deployed contract.

**Reply by 0xCalibur** [reply]
forge-deploy

**Comment by ankurdubey521** [comment]
- Better documentation on multi-chain deployments would be great, didn't even know that was supported, I came across it in a random tweet.
- Also a way to manage configuration for scripts would be nice. I find json too clunky, first class support for yaml would be nice.
- Built in support for deployment via standard create2/create3 factories.
- Not sure if this would be useful, but maybe native helper functions to do transactions via 4337 instead of EOAs

**Reply by PatrickAlphaC** [reply]
create2 is built in with `{salt}`. Have you deployed using create2 in solidity before?

**Reply by ankurdubey521** [reply]
Well yes, I have deployed using create2 before, but what about it?

**Comment by Lavishq** [comment]
well, i would prefer to use script with python but in foundry 
solidity isn't a scripting language
but we are programmer we manage with what works the best

**Comment by allwin199** [comment]
I would like to see a script that encrypts private key for deployment.
